torch/csrc/jit/interpreter.cpp

```diff
@@ -412,6 +412,21 @@ struct CodeImpl {
     for (auto node : block->nodes()) {
       const auto& source_location = node->getSourceLocation();
       switch (node->kind()) {
+        case prim::profile:
+        {
+          Operation op = [this, node](Stack& stack) -> int
+          {
+            IValue t;
+            pop(stack, t);
+
+            if (t.isTensor())
+            {
+                this->profiling_record_->record(t, static_cast<size_t>(node->i(attr::value)));
```

Comment:
casting to size_t for a unique ID seems funky when we have unique addressable `Value` nodes

Reply:
They have very nice properties, though. They are consecutive. They should remain valid after graph copies if the ids in an original graph were consecutive. They should be serializable and deserializable. They are more versatile in a sense we could use them in IR as well, and in different places, etc